File: test/Data.Analyzers.Vsix.Test/VisualBasic/ModelMemberAttributeRequiresArgument.vb
Imports DevZest.Data.Annotations

Public Class ModelMemberAttributeRequiresArgument
    Inherits Model

    Shared Sub New()
        RegisterColumn(Function(x As ModelMemberAttributeRequiresArgument) x.ID)
    End Sub

    Private m_ID As _Int32
    <DbColumn>
    Public Property ID As _Int32
        Get
            Return m_ID
        End Get
        Private Set
            m_ID = Value
        End Set
    End Property

End Class

File: test/Data.Analyzers.Vsix.Test/CSharp/ModelMemberAttributeRequiresArgument.cs
using DevZest.Data.Annotations;

namespace DevZest.Data.Analyzers.Vsix.Test.CSharp
{
    public class ModelMemberAttributeRequiresArgument : Model
    {
        static ModelMemberAttributeRequiresArgument()
        {
            RegisterColumn((ModelMemberAttributeRequiresArgument _) => _.Id);
        }

        [DbColumn]
        public _Int32 Id { get; private set; }
    }
}
